File: C#/Calculadora.cs
using System;

namespace ICalculadora
{
    class Calculadora
    {
        static void Main(string[] args)
        {
            Console.WriteLine("- Calculadora -");

            String opcaoCalculadora; do
            {
                Console.Write("Insira o primeiro valor: ");
                float primeiroValor = float.Parse(Console.ReadLine());
                Console.Write("Insira o segundo valor: ");
                float segundoValor = float.Parse(Console.ReadLine());

                // Criação do objeto utilizado no switch case.
                Calculadora calculadora = new Calculadora();

                String opcaoOperacoes; do
                {
                    Console.WriteLine("\nInsira:");
                    Console.WriteLine("+ para Somar");
                    Console.WriteLine("- para Subtrair");
                    Console.WriteLine("x para Multiplicar");
                    Console.WriteLine("/ para Dividir");
                    Console.WriteLine("S para Sair");

                    Console.Write("\nOpção: ");
                    opcaoOperacoes = Console.ReadLine();

                    switch (opcaoOperacoes)
                    {
                        case "+":
                            int resultado = calculadora.Somar(primeiroValor, segundoValor);
                            Console.WriteLine("\nResultado: " + resultado);
                            break;

                        case "-":
                            resultado = calculadora.Subtrair(primeiroValor, segundoValor);
                            Console.WriteLine("\nResultado: " + resultado);
                            break;

                        case "X":
                        case "x":
                            resultado = calculadora.Multiplicar(primeiroValor, segundoValor);
                            Console.WriteLine("\nResultado: " + resultado);
                            break;

                        case "/":
                            // Variável foi definida como "resultadoFloat" porque ela não pode ser
                            // do tipo int ao receber um retorno do tipo float da função Dividir().
                            float resultadoFloat = calculadora.Dividir(primeiroValor, segundoValor);
                            Console.WriteLine("\nResultado: " + resultadoFloat);
                            break;

                        case "S":
                        case "s":
                            Console.WriteLine("\nOperação encerrada.");
                            break;

                        default:
                            Console.WriteLine("\nComando não reconhecido.");
                            break;
                    }

                } while (opcaoOperacoes != "S" && opcaoOperacoes != "s");

                Console.WriteLine("\nInsira:");
                Console.WriteLine("Qualquer caracter para calcular novamente");
                Console.WriteLine("S para encerrar calculadora");

                Console.Write("\nOpção: ");
                opcaoCalculadora = Console.ReadLine();

            } while (opcaoCalculadora != "S" && opcaoCalculadora != "s");

            // Mensagem exibida após o fim do programa.
            Console.WriteLine("\nCalculadora encerrada.");
        }

        #region Funções
        int Somar ( float primeiroValor, float segundoValor )
        {
            return Convert.ToInt32( primeiroValor + segundoValor );
        }

        int Subtrair (float primeiroValor, float segundoValor )
        {
            return Convert.ToInt32(primeiroValor - segundoValor);
        }

        int Multiplicar (float primeiroValor, float segundoValor )
        {
            return Convert.ToInt32(primeiroValor * segundoValor);
        }
        float Dividir (float primeiroValor, float segundoValor )
        {
            return primeiroValor / segundoValor;
        }

        #endregion
    }
}

File: Java/calculadora.java
package fontes;
import java.util.Scanner;
public class calculadora 
{
    private double n1,n2,res;
    private Scanner s;
    
    public calculadora()
    {
        s = new Scanner(System.in);
        n1=0;
        n2=0;
        res=0;
    }
    
    public void lerDados()
    {
        System.out.println("Informe n1..:");
        n1 = s.nextDouble();
        System.out.println("Informe n2..:");
        n2 = s.nextDouble();
    }
    public void somar()
    {
        res = n1 + n2;
    }
    public void subtrair()
    {
        res = n1 - n2;
    }
    public void multiplicar()
    {
        res = n1 * n2;
    }
    public void dividir()
    {
        res = n1 / n2;
    }
    public void imprimir()
    {
        System.out.println("Resultado: "+res);
    }
    public static void main(String args[])
    {
        calculadora obj = new calculadora();
        obj.lerDados();
        obj.somar();
        obj.imprimir();
        obj.subtrair();
        obj.imprimir();
        obj.multiplicar();
        obj.imprimir();
        obj.dividir();
        obj.imprimir();
    }
}
